components/Banner/examples/BannerTypicalUseExampleViewController.m

```diff
@@ -160,199 +205,137 @@ + (NSDictionary *)catalogMetadata {
 #pragma mark - Example handlers
 
 - (void)showSingleLineStyleBanner {
-  if (self.bannerViewContainer) {
-    [self.bannerViewContainer removeFromSuperview];
+  if (self.bannerView) {
+    [self.bannerView removeFromSuperview];
   }
-  // Set up banner view container
-  UIView *bannerViewContainer = [[UIView alloc] initWithFrame:CGRectZero];
-  [self.view addSubview:bannerViewContainer];
-  bannerViewContainer.backgroundColor = [UIColor whiteColor];
-  self.bannerViewContainer = bannerViewContainer;
 
   MDCBannerView *bannerView = [[MDCBannerView alloc] init];
-  bannerView.image = [UIImage imageNamed:@"Email"];
+  bannerView.preferredContentWidth = exampleBannerContentWidth;
   bannerView.text = exampleShortText;
+  bannerView.backgroundColor = self.colorScheme.surfaceColor;
+  [self.view addSubview:bannerView];
+  self.bannerView = bannerView;
 
-  MDCButton *button = [bannerView.buttons firstObject];
+  MDCButton *button = [[MDCButton alloc] initWithFrame:CGRectZero];
   [button setTitle:@"DISMISS" forState:UIControlStateNormal];
-  UIFont *buttonFont = [MDCTypography body2Font];
-  [button setTitleFont:buttonFont forState:UIControlStateNormal];
-  [button setTitleFont:buttonFont forState:UIControlStateHighlighted];
   [button setTitleColor:self.colorScheme.primaryColor forState:UIControlStateNormal];
+  [button sizeToFit];
   button.backgroundColor = self.colorScheme.surfaceColor;
+  NSMutableArray *buttons = [NSMutableArray arrayWithObject:button];
+  bannerView.buttons = buttons;
 
-  [self.bannerViewContainer addSubview:bannerView];
   [button addTarget:self
                 action:@selector(dismissBanner)
       forControlEvents:UIControlEventTouchUpInside];
-  CGSize bannerViewSize =
-      [bannerView sizeThatFits:CGSizeMake(exampleBannerContentSingleLineWidth, CGFLOAT_MAX)];
-  bannerView.frame = CGRectMake(0.0f, 0.0f, bannerViewSize.width, bannerViewSize.height);
+  [bannerView sizeToFit];
 
   // Adjust bannerViewContainer's frame
   CGFloat topAreaInset = 0.0f;
   if (@available(iOS 11.0, *)) {
     topAreaInset = self.view.safeAreaInsets.top;
   }
-  self.bannerViewContainer.frame = CGRectMake(
-      0.0f, topAreaInset, [UIScreen mainScreen].bounds.size.width, bannerViewSize.height);
-  bannerView.center = CGPointMake(self.bannerViewContainer.frame.size.width / 2,
-                                  self.bannerViewContainer.frame.size.height / 2);
+
+  bannerView.frame =
+      CGRectMake(0.0f, topAreaInset, bannerView.frame.size.width, bannerView.frame.size.height);
```

Comment:
Please always either standardize frame/bounds using `CGRectStandardize()` and then use the result, or else please use `CGRectGetWidth()` and `CGRectGetHeight()`

Reply:
Addressed in 7dc9208cbfc2e2fe0d4c183c460c8b32414b4f27.